integration/spark/src/main/scala/org/apache/spark/sql/CarbonDataFrameWriter.scala

```diff
@@ -74,6 +74,9 @@ class CarbonDataFrameWriter(sqlContext: SQLContext, val dataFrame: DataFrame) {
       case decimal: DecimalType => s"decimal(${decimal.precision}, ${decimal.scale})"
       case BooleanType => CarbonType.BOOLEAN.getName
       case BinaryType => CarbonType.BINARY.getName
+      case ArrayType(elementType, _) => sparkType.simpleString
+      case StructType(fields) => sparkType.simpleString
+      case MapType(keyType, valueType, _) => sparkType.simpleString
```

Comment:
Long string columns (CarbonType.VARCHAR) also seems to be missing, can you test and add it also if required?

Reply:
There is no varchartype in spark.
https://spark.apache.org/docs/latest/sql-ref-datatypes.html

I have rework the testcase, All datatype are tested. please have a check

Reply:
yes, for spark all string is varchar itself. 
But for carbon if we mention long_string_columns in table properties we change data type from carbonType.string --> carbonType.varchar
for dataframe I think options("long_string_columns"="c1,c2") is not yet supported. So, current changes are ok